script.program.homeassistant/addon.py

```diff
@@ -0,0 +1,463 @@
+					
+
+			elif domain == 'scene':
+				icon = os.path.join(imgIconResourcePath,'scene.png')
+		
+			elif domain == 'script':
+				if 'last_triggered' in response[entity]['attributes']:
+					if response[entity]['attributes']['last_triggered'] is not None:
+						label = label  + markup + __addon__.getLocalizedString(30022) + str(parse_dateTime(response[entity]['attributes']['last_triggered'])) + '[/LIGHT]'
+
+			elif domain == 'sensor':
+				label = label + markup + entity_state + " "
+				if 'device_class' in response[entity]['attributes']:
+					if 'battery' in response[entity]['attributes']['device_class']:
+						icon = os.path.join(imgIconResourcePath,'battery.png')
+					elif 'connectivity' in response[entity]['attributes']['device_class']:
+						if entity_state == 'on':
+							icon = os.path.join(imgIconResourcePath,'connectivity.png')
+						else:
+							icon = os.path.join(imgIconResourcePath,'connectivity_off.png')
+					elif 'humidity' in response[entity]['attributes']['device_class']:
+						icon = os.path.join(imgIconResourcePath,'humidity.png')
+					elif 'motion' in response[entity]['attributes']['device_class']:
+						if 'last_changed' in response[entity]:
+							if response[entity]['last_changed'] is not None:
+								log(str(parse_dateTime(response[entity]['last_changed'])))
+								label = label + "Last changed: " + str(parse_dateTime(response[entity]['last_changed']))
+						icon = os.path.join(imgIconResourcePath,'motion.png')
+					elif 'opening' in response[entity]['attributes']['device_class']:
+						if entity_state == 'on':
+							icon = os.path.join(imgIconResourcePath,'opening.png')
+						else:
+							icon = os.path.join(imgIconResourcePath,'opening_off.png')
+					elif 'power' in response[entity]['attributes']['device_class']:
+						if entity_state == 'on':
+							icon = os.path.join(imgIconResourcePath,'power.png')
+						else:
+							icon = os.path.join(imgIconResourcePath,'power_off.png')
+					elif 'temperature' in response[entity]['attributes']['device_class']:
+						icon = os.path.join(imgIconResourcePath,'thermometer.png')
+					elif 'window' in response[entity]['attributes']['device_class']:
+						if entity_state == 'on':
+							icon = os.path.join(imgIconResourcePath,'window.png')
+						else:
+							icon = os.path.join(imgIconResourcePath,'window_off.png')
+				else:
+					icon = os.path.join(imgIconResourcePath,'sensor.png')
+
+				if 'unit_of_measurement' in response[entity]['attributes']:
+					label = label + response[entity]['attributes']['unit_of_measurement']
+				label = label +  '[/LIGHT]'
+
+			elif domain == 'switch':
+				if entity_state == 'off':
+					icon = os.path.join(imgIconResourcePath,'switch_off.png')
+
+			elif domain == 'vacuum': # Each for Start / Stop / Return to base / Locate
+				label = '[B]' + label + '[/B][CR][LIGHT]State: ' + response[entity]['attributes']['status'] + ' - Battery: ' + str(response[entity]['attributes']['battery_level']) + '[/LIGHT]' 
```

Comment:
sorry, any strings displayed on the interface have to be localized. So Battery and State has to be moved to strings.po